src/NodeParser/MappedTypeNodeParser.ts

```diff
@@ -47,8 +48,18 @@ export class MappedTypeNodeParser implements SubNodeParser {
             // Key type widens to `string`
             const type = this.childNodeParser.createType(node.type!, context);
             const resultType = type === undefined ? undefined : new ObjectType(id, [], [], type);
-            if (resultType && constraintType instanceof AnnotatedType) {
-                const annotations = constraintType.getAnnotations();
+            if (resultType) {
+                let annotations;
+
+                if (constraintType instanceof AnnotatedType) {
+                    annotations = constraintType.getAnnotations();
+                } else if (constraintType instanceof DefinitionType) {
+                    const type = constraintType.getType();
+                    if (type instanceof AnnotatedType) {
+                        annotations = type.getAnnotations();
+                    }
+                }
```

Comment:
Here is the critical change. I am not familiar enough with the object model so perhaps I approached this wrong. This does give the correct result in this case, but perhaps you would do it differently?